src/agent/state.js

```diff
@@ -286,32 +287,44 @@ StateResolver.prototype.isPathInNodeModulesDirectory_ = function(path) {
   return path.indexOf('node_modules') === 0;
 };
 
-StateResolver.prototype.resolveFrame_ = function(frame, resolveVars) {
-  var noArgs = [{
+StateResolver.prototype.produceArgumentsNotAvailableMessage = function () {
+  return {
     name: 'arguments_not_available',
     varTableIndex: ARG_LOCAL_LIMIT_MESSAGE_INDEX
-  }];
-  var noLocals = [{
+  };
+};
+
+StateResolver.prototype.produceLocalsNotAvailableMessage = function () {
+  return {
     name: 'locals_not_available',
     varTableIndex: ARG_LOCAL_LIMIT_MESSAGE_INDEX
-  }];
-  var args = this.extractArgumentsList_(frame);
-  var locals = resolveVars ? this.resolveLocalsList_(frame, args) : noLocals;
-  if (isEmpty(locals)) {
-    locals = noLocals;
-  }
-  if (semver.satisfies(process.version, '<1.6')) {
-    // If the node version is over 1.6 we do not read the frame arguments since
-    // the values produced by the frame for the arguments may contain inaccurate
-    // values. If the version is lower than 1.6, though, the frame's argument
-    // list can be relied upon to produce accurate values for arguments.
-    args = resolveVars && (!isEmpty(args)) ? this.resolveArgumentList_(args) :
-      noArgs;
+  };
+};
+
+StateResolver.prototype.resolveFrame_ = function(frame, underFrameCap) {
+  var args = [];
+  var locals = [];
+  if (!underFrameCap) {
+    args.push(this.produceArgumentsNotAvailableMessage());
```

Comment:
Since there is no other use of a constant object building method `produceArgumentsNotAvailable`, just inline the object here and get rid of the function.